app/src/main/java/org/akvo/flow/activity/MapActivity.java

```diff
@@ -19,40 +19,134 @@
 
-public class MapActivity extends BackActivity implements RecordListListener {
+public class MapActivity extends BackActivity implements OnMapReadyCallback,
+        LoaderManager.LoaderCallbacks<SurveyedLocale> {
+
+    public static final int MAP_ZOOM_LEVEL = 10;
+
+    private SurveyedLocale datapoint;
+    private String datapointId;
+    private GoogleMap map;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        setContentView(R.layout.map_activity);
-
-        String datapointId = getIntent().getStringExtra(ConstantUtil.SURVEYED_LOCALE_ID);
+        setContentView(R.layout.activity_map);
+        SupportMapFragment mapFragment = (SupportMapFragment) getSupportFragmentManager()
+                .findFragmentById(R.id.map);
+        mapFragment.getMapAsync(this);
+        datapointId = getIntent().getStringExtra(ConstantUtil.SURVEYED_LOCALE_ID);
+        loadItem();
+    }
 
-        SurveyDbAdapter db = new SurveyDbAdapter(this).open();
-        SurveyedLocale datapoint = db.getSurveyedLocale(datapointId);
-        db.close();
+    private void loadItem() {
+        getSupportLoaderManager().restartLoader(0, null, this);
+    }
 
+    private void updateTitle() {
         if (datapoint != null) {
             setTitle(datapoint.getDisplayName(this));
         }
+    }
 
-        getSupportFragmentManager()
-                .beginTransaction()
-                .replace(R.id.content_frame, MapFragment.newInstance(null, datapointId))
-                .commit();
+    @Override
+    public void onMapReady(GoogleMap googleMap) {
+        this.map = googleMap;
+        googleMap.setMyLocationEnabled(true);
+        loadItem();
+    }
+
+    private void addDataPointMarker() {
+        if (map != null) {
+            map.clear();
+            if (isDataPointLocationAvailable()) {
+                map.addMarker(new MarkerOptions()
+                        .position(new LatLng(datapoint.getLatitude(), datapoint.getLongitude()))
+                        .title(datapoint.getDisplayName(this))
+                        .snippet(datapoint.getId()));
+            }
+        }
+    }
+
+    private boolean isDataPointLocationAvailable() {
+        return datapoint != null && datapoint.getLatitude() != null
+                && datapoint.getLongitude() != null;
+    }
+
+    /**
+     * Center the map in the given record's coordinates. If no record is provided,
+     * the user's location will be used.
+     */
+    private void centerMap() {
+        if (isDataPointLocationAvailable()) {
+            centerMapOnDataPoint(datapoint);
+        } else {
+            centerMapOnUserLocation();
+        }
+    }
+
+    private void centerMapOnUserLocation() {
+        LatLng position = null;
+        LocationManager manager = (LocationManager) getSystemService(Context.LOCATION_SERVICE);
+        Criteria criteria = new Criteria();
+        criteria.setAccuracy(Criteria.ACCURACY_FINE);
```

Comment:
That is a very valid point indeed, i just copied the code that was already there but since it is the user location that we display here it would be totally fine to have coarse location which is faster and uses less battery, I have changed it.